.ci/heartbeat-synthetics.groovy

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env groovy
```

Comment:
Is this particular pipeline required? The current CI pipeline does not use it, so I wonder if the packaging pipeline should be the one to be used instead?

Reply:
@kuisathaverat would be the best one to ping here since he added that to this PR. However, I do believe it's essential (today) to make the custom synthetics image based off this branch.

Reply:
since the x-pack/heartbeat is build in package pipeline, I do not think we need the groovy file. the heartbeat Docker image will be the new synthetics  docker image. this pipeline is only to change the name to the Docker image to `...-synthetics`